server/test/deckTest.js

```diff
@@ -26,7 +107,129 @@ describe("Decks", function () {
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.property("name");
+        res.body.name.should.equal(testDeckName);
+        done();
+      });
+  });
+
+  it("GET Error, get deck by invalid ID", (done) => {
+    chai
+      .request(server)
+      .get("/decks/" + invalidId)
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql(invalidIdErrorMessage);
+        done();
+      });
+  });
+});
+
+describe("PATCH Deck name", function () {
+  it("PATCH, update test deck name", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + testDeckId)
+      .send({ name: "Spanish Vocab 1" })
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.a("object");
+        res.body.should.have.property("name").eql("Spanish Vocab 1");
+        done();
+      });
+  });
+});
+
+describe("PATCH Deck name", function () {
+  it("PATCH, update test deck name with special character", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + testDeckId)
+      .send({ name: "Spanish Vocabs!" })
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.a("object");
+        res.body.should.have.property("name").eql("Spanish Vocabs!");
+        done();
+      });
+  });
+
+  it("Revert for future test purposes", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + testDeckId)
+      .send({ name: testDeckName })
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.a("object");
+        res.body.should.have.property("name").eql(testDeckName);
+        done();
+      });
+  });
+
+  it("PATCH, error update deck with invalid id", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + invalidId)
+      .send({ name: "failed patch" })
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql(invalidIdErrorMessage);
+        done();
+      });
+  });
+
+  it("PATCH, error update deck with invalid whitespace name", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + testDeckId)
+      .send({ name: "   " })
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql("Valid name required");
+        done();
+      });
+  });
+
+  it("PATCH, error update deck with invalid special character name", (done) => {
+    chai
+      .request(server)
+      .patch("/decks/" + testDeckId)
+      .send({ name: ">!" })
+      .end((err, res) => {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql("Valid name required");
+        done();
+      });
+  });
+});
```

Comment:
What if the input is an empty string? (no white space)

Reply:
tested in line 60

Reply:
I thought an empty string for PATCH would be somewhat different, but if not, that's okay.